### DISCUSSION TITLE
Visible whitespace tabs (\t)?

### DISCUSSION THREAD

**Discussion by KenMacD** [kovidgoyal/kitty#5686] [Q&A]
Hello, I know a lot of tools have options for showing tabs, but I was wondering if there was any support at the terminal level.

From a grep of the source I see tabs are included, along with spaces, as using the `BLANK_FONT`, so my guess would be that it's not currently possible. Am I wrong? If it isn't possible is there any reason drawing something for tabs wouldn't work?

**Comment by kovidgoyal** [comment]
It's not possible, and its not as easy as "drawing something for tabs".
When an application sends tabs to the terminal it does so in the
expectation of some behavior, breaking that expectation can break
applications.

**Comment by KenMacD** [comment]
Thank you. I suppose that makes sense. I was thinking the 'something' would be the same width as a tab shifts, so formatting wouldn't change. Maybe something like a very light variable width `|---|`.

I'm currently using an alias that does a `sed` to replace tabs in output, but because it doesn't know the current cursor position it can't really align things the way I might expect. It does get me more than 80% the way there though, so I'll probably stick with that.

**Reply by kovidgoyal** [reply]
so tab widths in a terminal are application configurable, they are not fixed size. Indeed temrinals have a concept of tabstops which can be arbitrary columns they dont even need to all be the same size. You could fairly easily patch kitty to draw something like a dot or similar on any cell that has tabs but this would look pretty ugly since you would get eight dots for every tab (assuming default tabstops). To get what you want would require identifying contiguous runs of cells with tab in them and then rendering something special there.  This is also technically doable, but rather more effort than its worth and it certainly wont change how things are aligned, it would just give you a visual indication of where tabs are used instead of spaces for alignment. Actually changing the alignment would break things.
